File: OOB class/proj2_java/src/symulator/rosliny/guarana.java
package symulator.rosliny;

import symulator.Organizm;
import symulator.Roslina;
import symulator.Swiat;

public class guarana extends Roslina {


    public guarana(Swiat swiat, int ox, int oy,  int wiek, int BOARD_WIDTH, int BOARD_HEIGHT) {
        super(swiat, ox, oy, wiek, BOARD_WIDTH, BOARD_HEIGHT);

        this.nazwa = "Guarana";
        this.inicjatywa = 0;
        this.sila = 0;
        this.znak = 'G';
        swiat.setJesliWolnePole(ox, oy, znak);
    }

    @Override
    public void zasiej(int x, int y) {
        swiat.addOrganizm(new guarana(swiat,x, y, swiat.get_and_incrementKtoryUrodzony(),BOARD_WIDTH,BOARD_HEIGHT));
    }

    @Override
    public int kolizja(Organizm napotkany_org) {
        napotkany_org.zwiekszSile(3);
        return WIN;
    }


}

File: OOB class/proj1_cpp/guarana.cpp
#include "guarana.h"
#include "Swiat.h"

guarana::guarana(short x, short y, Swiat* swiat, short wiek) {
	this->wiek = wiek;
	this->nazwa = "Guarana";
	this->inicjatywa = 0;
	this->sila = 0;
	this->znak = 'G';
	this->polozenie[OX] = x;
	this->polozenie[OY] = y;
	this->swiat = swiat;
	swiat->setJesliWolnePole(x, y, znak);
}

void guarana::zasiej(short x, short y) {
	swiat->addOrganizm(new guarana(x, y, this->swiat, swiat->get_and_incrementKtoryUrodzony()));
}

int guarana::kolizja(Organizm* napotkany_org) {
	napotkany_org->zwiekszSile(3);
	return WIN;
}
